PRAGMA's MissingEntry: Suggest CALL might be an option

carlopi opened the pull request:
Unsure if worth adding, but given over time some PRAGMA have become CALL, and more will come, maybe it make sense?

Example:
```sql
PRAGMA enable_logging(level='trace', storage='stdout');
```
goes from
```
Catalog Error:
Pragma Function with name enable_logging does not exist!
Did you mean "enable_profiling"?
```
to
```
Catalog Error:
Pragma Function with name enable_logging does not exist!
Did you mean "enable_profiling"?
Pragma Function might be available also as table function statements (eg: `CALL ...();`)
```

carlopi commented:
Thanks for the feedback, I don't see exactly how to make it work with limited changes in BindPragma, there is a line like:
```
auto &entry = Catalog::GetEntry<PragmaFunctionCatalogEntry>(context, INVALID_CATALOG, DEFAULT_SCHEMA, info.name);
```
and it will throw directly from deeper in the call stack there, and it's probably not worth (due to excessive surface it touches) to change the code there.

I can also see just saying "don't use pragma" on every error is not OK.

Mytherin commented:
```cpp
auto entry = Catalog::GetEntry<PragmaFunctionCatalogEntry>(context, INVALID_CATALOG, DEFAULT_SCHEMA, info.name, OnEntryNotFound::RETURN_NULL);
if (!entry) {
    auto table_entry = Catalog::GetEntry<TableFunctionCatalogEntry>(context, INVALID_CATALOG, DEFAULT_SCHEMA, info.name, OnEntryNotFound::RETURN_NULL);
    if (table_entry) {
        // there is a table entry with the same name
    }
    // rebind to throw exception
    entry = Catalog::GetEntry<PragmaFunctionCatalogEntry>(context, INVALID_CATALOG, DEFAULT_SCHEMA, info.name, OnEntryNotFound::THROW_EXCEPTION);
}

carlopi committed:
PRAGMA's MissingEntry: Suggest CALL might be an option

Basically @Mytherin's suggestion, see conversation at https://github.com/duckdb/duckdb/pull/18815

carlopi commented:
This should also be ready to go in.

Mytherin commented:
Thanks!
